pkgs/hdf5/hdf5.yaml

```diff
@@ -18,15 +18,16 @@ build_stages:
   bash: |
     export CC=${MPICC}
 
-- when: machine == 'CrayXE6'
+- when: machine == 'CrayXE6' or machine == 'CrayXC30'
   name: make_exe_builds_static
   handler: bash
   after: configure
   before: make
   bash: |
-    for f in {test,testpar,tools/*,perform,hl/*,hl/*/*}/Makefile 
+    for f in {src,test,testpar,tools/*,perform,hl/*,hl/*/*}/Makefile 
```

Comment:
@mbareford this causes a runtime problem with the hdf5 shared libraries when I test  on our XC30. The issue is that the '-static' option added in the original patch was  only intended for  the directories in the original list so it  breaks the shared  libs at runtime when it is also applied  to src. I made a branch fix_hdf5_sed that applies your sed fix in a second  loop and verified  that it  works. Could you merge that into your  branch and test on Archer? 
/cc @tridelat @adimako